products/bridge/smart-contracts/contracts/ERC20Bridge.sol

```diff
@@ -69,10 +102,8 @@ contract ERC20Bridge is Bridged {
         emit Started(token, owner, value);
     }
 
-    event Succeeded();
-    event Failed(string);
-
     function finish(
+        uint targetChainId,
```

Comment:
You're correct, this is no longer needed as ERC20Bridge inherits the `BridgedTwin`, so it is always the `twinChainId`

Reply:
The arguments to the `callback` function needs to include `targetChainId`, so the non-twin bridges can know where the messages come from if they ever need it for housekeeping

Reply:
I agree with you that in general, bridged contracts can request cross-chain calls to multiple target chains and they need to know from which one they're receiving a response (even though they could figure it out based on the `nonce`, assuming that they must store what they requested on which target chain for every nonce anyway).

In this specific case, because we agreed that the `ERC20Bridge` will have a hardwired twin chain and can't receive responses from any other chain id than the one returned by `twinChainId()`, I thought passing the target chain id to the `finish` function was redundant...

Reply:
This would also allow the querying you had mentioned to retrieve which chain the resume came from by checking against the relayer

Reply:
```solidity
    // From this:
    // targetChainId => caller => nonce
    mapping(uint => mapping(address => uint)) nonces;
    // To this:
    // caller => nonce
    mapping(address => int) nonces
    // caller => nonce => targetChainId
    mapping(address => mapping(int => int)) nonceTargetChainIds
```

Reply:
As concluded in our call, you can remove the `targetChainId` from the list of arguments in the `finish` function and let the "multichain" bridged contracts keep track of the mapping between the `nonce` and the `targetChainId`. This resolves the comment from my point of view.